### ISSUE TITLE
Per-target cooking?

### ISSUE THREAD

**Issue by norru** [vityafx/cargo-cook#80]
Hi,

I just found your cool cargo extension. I want to use it to package distinct distribution files for Windows and Linux.

What is the recommended mechanism to achieve this?

**Comment by vityafx** [comment]
Hi! I was not planning on having providing Windows compatibility because I don't have windows installed. However, definitely, such a project, if it has users, should allow working on Windows as well.

Right now I don't have any idea whether it works on Windows or not. If it does not, definitely something should be done to solve this, but simply I can't do it myself as I don't have Windows PC.

If someone makes a PR bringing Windows support, I would definitely accept it, with pleasure.

P.S. Actually there was an [attempt](https://github.com/vityafx/cargo-cook/pull/48) to do this but the contributor did not want to participate any longer, and also I can not test it myself, only do a review. So, if you are going to fix problems in this PR and make another one, I will merge it ASAP :)

**Comment by vityafx** [comment]
Sorry,  I guess I have misunderstood. Do you want to package via cargo-cook for Windows while running it under Linux?

**Comment by norru** [comment]
Yes.

I am building some software which has to run in Windows and Linux and I need to build two separate packages, one which should run in Windows and one which should run in Linux, in the same build.

I have solved the cross-compiling issues; I can manually generate two separate `.zip` files. I just want this to be automated.

**Comment by norru** [comment]
However, this is no longer urgent as I wrote a custom packager as a simple Rust executable which does exactly this.

**Comment by vityafx** [comment]
Well, I am not sure what the problem exactly you were struggling with. Is it something like making `cargo-cook` to build not one pack of archives but more than one perhaps for different purposes (like one for Linux, another for Windows, etc)?

**Comment by norru** [comment]
Yes, and those packages will contain a different set of assets (dll for Windows but not for Linux etc).

Perhaps having Cook.<Windows toolchain>.toml and Cook.<Linux toolchain>.toml would do?

**Comment by vityafx** [comment]
I think it would be better to specify just different packages. I'll definitely think about this. This sounds a nice idea to me.

**Comment by vityafx** [comment]
@norru Hi there! I've just added support for specifying custom recipes. Maybe this can help you:

```
cargo cook -r CookForLinux.toml
cargo cook -r CookForWindows.toml
```
